### ISSUE TITLE
empty-lines, false positives in presence of comments

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#84]
**Describe the bug**
The rule `empty-lines` warns on blocks starting or ending with comments

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following command line and configuration file:

Command line
```shell
./revive -config=test.toml -formatter friendly test.go
```

Configuration file `test.toml`
```toml
ignoreGeneratedHeader = false
severity = "warning"
confidence = 0.8
errorCode = 0
warningCode = 0

[rule.empty-lines]
```

Test file `test.go`

```go
// Test of empty-lines.

package fixtures

func w() {
	select {
	case <-time.After(dur):
		// TODO: Handle Ctrl-C is pressed in `mysql` client.
		// return 1 when SLEEP() is KILLed
	}
	return 0, false, nil
}

func x() {
	if tagArray[2] == "req" {
		bit := len(u.reqFields)
		u.reqFields = append(u.reqFields, name)
		reqMask = uint64(1) << uint(bit)
		// TODO: if we have more than 64 required fields, we end up
		// not verifying that all required fields are present.
		// Fix this, perhaps using a count of required fields?
	}

	if err == nil { // No need to refresh if the stream is over or failed.
		// Consider any buffered body data (read from the conn but not
		// consumed by the client) when computing flow control for this
		// stream.
		v := int(cs.inflow.available()) + cs.bufPipe.Len()
		if v < transportDefaultStreamFlow-transportDefaultStreamMinRefresh {
			streamAdd = int32(transportDefaultStreamFlow - v)
			cs.inflow.add(streamAdd)
		}
	}
}
```

**Expected behavior**

revive must not emit any warning.

**Logs**
Actual output:

```shell
  ⚠  empty-lines  extra empty line at the end of a block
  fixtures/test.go:7:2

  ⚠  empty-lines  extra empty line at the end of a block
  fixtures/test.go:18:3

  ⚠  empty-lines  extra empty line at the start of a block
  fixtures/test.go:24:16

⚠ 3 problems (0 errors, 3 warnings)

Warnings:
  3  empty-lines
```

**Comment by mgechev** [comment]
cc @gsamokovarov

**Comment by gsamokovarov** [comment]
The comments are tricking the rule here. I'll put more work in the comments between nodes detection and get back to you.
